src/com/haskforce/codeInsight/HaskellCompletionCacheLoader.scala

```diff
@@ -7,38 +7,26 @@ import com.haskforce.psi.{HaskellFile, HaskellPsiUtil}
 import com.intellij.AppTopics
 import com.intellij.codeInsight.lookup.LookupElement
 import com.intellij.openapi.application.ApplicationManager
-import com.intellij.openapi.components.ProjectComponent
 import com.intellij.openapi.editor.Document
 import com.intellij.openapi.fileEditor._
-import com.intellij.openapi.project.Project
+import com.intellij.openapi.project.{Project, ProjectManagerListener}
 import com.intellij.openapi.util.Computable
 import com.intellij.openapi.vfs._
 import com.intellij.psi.{PsiFile, PsiManager}
 
 import scala.annotation.tailrec
 
 /** Loads the completion cache for any Haskell files we have open. */
-class HaskellCompletionCacheLoader(project: Project) extends ProjectComponent {
+class HaskellCompletionCacheLoader extends ProjectManagerListener {
```

Comment:
There's also another issue. In the companion object there's this

```scala
object HaskellCompletionCacheLoader {
  def get(project: Project): HaskellCompletionCacheLoader = {
    project.getComponent(classOf[HaskellCompletionCacheLoader])
  }
```

That obviously won't work anymore. Need to find a better way to do this. The `cache` field needs to be scoped by project, which isn't quite addressed in this PR yet.

Reply:
no problem. I'll surface it as a service. Then it is scoped and distributed to all projects

Reply:
I see this part was bundling state with the component. That has to be refactored into a separate service, that will manage and store the cache. 
I am only voicing this so you understand the steps I am taking

Reply:
In the future versions of Intellij that has to be separated. 
pure Morphisms can be distributed as services with functions that can be overridden by 3rd parties depending on this plugin. If you choose to. 
IntelliJ also allows them to contain mutable variables or state, but personally I'd like to keep them apart, as this allows one to reason over simpler.